### PR TITLE
Swat/feature/merging master into daemon mode

### PR DESCRIPTION
**Pull request by sshankar19**

### CODE REVIEW

**Review by nikmash — DISMISSED** [review]

**Review by nsaenz7 — APPROVED** [review]

**Review by nsaenz7 — APPROVED** [review]

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
-# 4.0.0-beta
+## 4.0.0
 * Enabling daemon applications to service their own endpoints.
+## 3.1.0
+* Updating supertype to 3.1.0, adding a public api to allow customization of supertype's logging functionality
```
Comment (hackerrdave) [inline HISTORY.md:4]: updating supertype to 3.1.0 is occurring in Amorphic v4.0.0 not Amorphic v3.1.0 correct?

**Review by hackerrdave — CHANGES_REQUESTED** [review]

File: HISTORY.md (hunk 1 above)
Comment (sshankar19) [reply HISTORY.md:4]: No, this happened already in Amorphic 3.1, and that was already published. I'm just porting those changes back into Amorphic 4.0

Comment (nikmash) [reply HISTORY.md:4]: Hilariously confusing because it's both amorphic 3.1.0 and supertype 3.1.0

Comment (hackerrdave) [reply HISTORY.md:4]: what's the merging strategy for the various branches/versions?

Comment (hackerrdave) [reply HISTORY.md:4]: hard to know how to review this PR w/o understanding our branching/versioning

Comment (sshankar19) [reply HISTORY.md:4]: After this MR, we will merge this branch into master, and then publish 4.0. Thought about maintaining two different versions of amorphic and that's too much work.

**Review by hackerrdave — APPROVED** [review]

**Pull request merged**